kudu/src/main/java/org/apache/apex/malhar/kudu/IncrementalStepScanInputOperator.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.kudu;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.kudu.client.ExternalConsistencyMode;
+import org.apache.kudu.client.SessionConfiguration;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * <p>A Kudu input operator that provides the functionality of scanning a Kudu table by incrementing
+ * a column in steps as specified in the configuration. Each iteration scans the table as given by the
+ *  SQL expression in the properties file. Other aspects like connection details are also read from the properties file.
+ * </p>
+ *
+ * <p>
+ *   Here is an example property file format.
+ *   Place a file named kuduincrementalstepscaninputoperator.properties anywhere in the classpath ( say by
```

Comment:
What's the advantage using the configuration file vs. configuring individual properties on the operator?

Reply:
Configuration files would mean lesser code in the application class. If the template query were to be changed and redeployed , it does not mean a code change and instead replace the properties file in the class path. However I see what you mean, and hence made changes to support both models. There are additional getters/setters and the default constructor does not throw an exception if the properties file is not in the classpath.